packages/fx-core/src/component/generator/generator.ts

```diff
@@ -204,10 +211,10 @@ export async function sampleDefaultOnActionError(
         await fs.rm(context.destination, { recursive: true });
       }
       if (error instanceof BaseComponentInnerError) throw error.toFxError();
-      else if (error.message.includes("403")) {
-        throw new DownloadSampleApiLimitError(context.url!).toFxError();
+      else if (isApiLimitError(error)) {
+        throw new DownloadSampleApiLimitError(convertToUrl(context.sampleInfo!), error).toFxError();
```

Comment:
What's different between context.url and convertToUrl(context.sampleInfo) ?

Reply:
There are some changes for downloading sample https://github.com/OfficeDev/TeamsFx/pull/10238 to support '/' in branch, so now the context.url will be empty for downloading sample and we need to concat sample info to get the url.